source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -130,6 +130,19 @@ void AuthenticatorImpl::verify(Http::HeaderMap& headers, Tracing::Span& parent_s
     return;
   }
 
+  // By default, this filter extracts JWT token from Authorization header and
+  // access_token query parameter. A request may have multiple JWT tokens, and will be
+  // forwarded to the backend if one of the tokens is good. It poses a security risk:
+  // a hacker can put a good token in the query parameter and an invalid one in
+  // the Authorization header. Envoy will forward the request to the backend,
+  // and the backend will use the bad token in Authorization header.
+  // This check will patch such security hole.
```

Comment:
is this changing to just reject the request if it has multiple token even when all of them are valid? Can you make it a bit clear in the comment of how is this check patching the security hole?

Also do you need to update the documentation as well? https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/jwt_authn_filter#default-extract-location

Reply:
I changed my implementation.  I think if a request has multiple tokens, all of them should be valid. This will prevent such security hole.